beta-src/src/utils/state/startNewOrder.ts

```diff
@@ -1,28 +1,25 @@
 import { current } from "@reduxjs/toolkit";
+import UIState from "../../enums/UIState";
 import NewOrderPayload from "../../interfaces/state/NewOrderPayload";
 import { GameCommand } from "../../state/interfaces/GameCommands";
 import getDataForOrder from "./getDataForOrder";
 import setCommand from "./setCommand";
 
 /* eslint-disable no-param-reassign */
 export default function startNewOrder(state, action: NewOrderPayload): void {
+  console.log("startNewOrder");
   const {
     order: { unitID: prevUnitID, type },
   } = current(state);
   if (prevUnitID && type !== "hold" && type !== "disband") {
-    const command: GameCommand = {
-      command: "NONE",
-    };
-    setCommand(state, command, "unitCommands", prevUnitID);
+    state.unitState[prevUnitID] = UIState.NONE;
   }
   delete state.order.type;
   const orderData = getDataForOrder(state, action.payload);
   state.order = orderData;
   const { unitID } = orderData;
+  // really the state should probably come directly from the order object
```

Comment:
The `order` in state is the current order in progress. To initiate a new order, the orderData here is information needed to start a new order process (NewOrderPayload).

Reply:
Yep. What I mean is that the `orders` and `ordersMeta` objects already have all the info needed to render the arrows and highlights, but I'm building a parallel set of state like `unitState` which is not ideal. But for my first pass at least I'm not going to resolve this, I'll just leave some parallel UI state that is similar to what was in the commands. But which describes the UI *state* rather than *state updates*.